Src/IronPython/Runtime/Binding/PythonProtocol.Operations.cs

```diff
@@ -1130,13 +1130,24 @@ static partial class PythonProtocol {
             SlotOrFunction.GetCombinedTargets(cmp, rcmp, out cmp, out rcmp);
 
             bool shouldWarn = false;
-            WarningInfo info = null;
+
+            // if the left operand is an instance of a built-in type or a new-style class, and the right operand is an instance of a proper subclass of that type or class
+            // and overrides the base’s __rop__() method, the right operand’s __rop__() method is tried before the left operand’s __op__() method.
+            PythonType xPythonType = MetaPythonObject.GetPythonType(xType);
+            PythonType yPythonType = MetaPythonObject.GetPythonType(yType);
+            bool opReverse = false;
+            if ((xPythonType.IsSystemType || !xPythonType.IsOldClass) && (yPythonType != xPythonType) && BindingHelpers.IsSubclassOf(yType, xType) && (rop != null)) {
```

Comment:
Minor optimization: `rop != null` check first so we don't have to check everything else

Reply:
Will do